core/trino-main/src/main/java/io/trino/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -109,15 +130,12 @@ private int getNewTaskCount()
                 .map(Optional::get)
                 .mapToLong(writerCount -> writerMinSizeBytes * writerCount)
                 .sum();
+        return writtenBytes >= minWrittenBytesToScaleUp;
+    }
 
-        // When there is a big data skewness, there could be a bottleneck due to the skewed workers even if most of the workers are not over-utilized.
-        // Check both, weighted output buffer over-utilization rate and average output buffer over-utilization rate, in case when there are many over-utilized small tasks
-        // due to fewer not-over-utilized big skewed tasks.
-        if ((isWeightedBufferFull() || isAverageBufferFull()) && (writtenBytes >= minWrittenBytesToScaleUp)) {
-            return 1;
-        }
-
-        return 0;
+    private boolean isUpperLimitExceeded()
```

Comment:
Do we need to have a separate method? I think it's better to have this condition directly in the above if condition

Reply:
ok